cm/heatlearn/heatlearn.py

```diff
@@ -0,0 +1,328 @@
+                }
+            },
+        )
+
+        df.append(pd.DataFrame(r.json()))
+    df = pd.concat(df, ignore_index=True)
+    if df.shape[0] > 0:  # there are NUTS3 with HDD
+        df["HDD"] = df["variables"].apply(lambda x: x["Heating degree days"])
+        df["HDD_nosummer"] = df.loc[
+            df.index.isin([0, 1, 2, 3, 4, 8, 9, 10, 11]), "HDD"
+        ]  # used for the model
+        return df.groupby("fid").sum().mean()[["HDD", "HDD_nosummer"]].values.tolist()
+    else:
+        # Find the NUTS3 code
+        r = requests.post(
+            API_URL,
+            headers=HEADERS,
+            json={
+                "parameters": {
+                    "data.ds_id": 0,
+                    "intersecting": "{}".format(polygon.wkt),
+                    "level": "{NUTS3}",
+                }
+            },
+        )
+        if r.json()[0]["fid"] == "CH013":  # this is the Canton of Geneva
+            CH013 = pd.read_csv("CH013_HDD.csv", index_col="time", parse_dates=True)
+            return CH013.loc[CH013.index.year == year, :].values.tolist()[0]
+        else:
+            raise ValueError("In this area Heating Degree Days are not available.")
+
+
+def heatlearn(geojson, raster_paths, tile_size=500, year=2020):
+    """Get heating demand from HeatLearn Model."""
+    if tile_size not in MODELS.keys():
+        raise ValueError(
+            "Only these tile sizes are possible: {}".format(", ".join(MODELS.keys()))
+        )
+
+    pixel_size = 2.5
+    start = time()
+
+    # Create boundary
+    geometries = []
+    for feature in geojson["features"]:
+        geometry = feature["geometry"]
+        geoshape = shape(geometry)
+        geometries.append(geoshape)
+    union_geometry = cascaded_union(geometries)
+    boundary = gpd.GeoSeries(union_geometry)
+    boundary = boundary.set_crs("EPSG:4326")
+    boundary = boundary.to_crs("EPSG:3035")
+    boundary = gpd.GeoDataFrame(boundary)
+    boundary = boundary.rename(columns={0: "geometry"}).set_geometry("geometry")
+
+    # Make sure that the clipping geometry does not leave any pixel out
+    boundary["geometry"] = boundary["geometry"].apply(
+        lambda x: wkt.loads(wkt.dumps(x, rounding_precision=0))
+    )
+    offset = boundary.total_bounds % pixel_size
+    boundary["geometry"] = boundary["geometry"].translate(
+        xoff=-offset[0], yoff=-offset[1]
+    )
+
+    # Make tiles
+    bounds = boundary.total_bounds
+    tiles = makeGrid(bounds, tile_size)
+    tiles = tiles.set_crs("EPSG:3035")
+    logging.info(boundary.geometry)
+    logging.info(tiles.total_bounds)
+    tiles = tiles.loc[tiles.intersects(boundary.iloc[0].geometry), :]
+    tiles = tiles.reset_index()
+    if tiles.shape[0] == 0:
+        raise ValueError("No tiles were created.")
+
+    # Prepare raster
+    if len(raster_paths) == 1:
+        raster_path = raster_paths[0]
+    else:
+        raise ValueError("Only a single raster is supported for now.")
```

Comment:
Make this check in the worker.py file, before calling the heatlearn function (and check all the other parameters there too).

Reply:
I think this should stay here. The `else` will take care of merging more rasters (TBD).